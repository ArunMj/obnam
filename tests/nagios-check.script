#!/bin/sh
# Copyright 2011  Lars Wirzenius
# 
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
# 
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
# 
# You should have received a copy of the GNU General Public License
# along with this program.  If not, see <http://www.gnu.org/licenses/>.

set -e

# Obnam settings.
opts="--quiet --no-default-config"

$SRCDIR/tests/backup

# Make sure enough time passes.
# FIXME: This should fake the timestamps rather than slow down the test suite.
# But there's no easy way to do that right now.
sleep 2

# Check there's a warning.
if ! ./obnam $opts -r "$DATADIR/repo" nagios-last-backup-age --warn-age=1s | 
    sed 's/last backup.*//'
then
    echo 'correctly returned non-zero exit code'
fi

# Check there's an error.
if ! ./obnam $opts -r "$DATADIR/repo" nagios-last-backup-age --critical-age=1s | 
    sed 's/last backup.*//'
then
    echo 'correctly returned non-zero exit code'
fi
