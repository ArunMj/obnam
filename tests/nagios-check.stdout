WARNING: backup is old.  
CRITICAL: backup is old.  
